contactaccessrights/js/test/controllers/access-rights-modal-ctrl_test.js

```diff
@@ -0,0 +1,179 @@
+define([
+  'common/angularMocks',
+  'access-rights/controllers/access-rights-modal-ctrl',
+  'access-rights/models/region',
+  'access-rights/models/location',
+  'access-rights/models/right'
+], function () {
+  'use strict';
+
+  describe('AccessRightsModalCtrl', function () {
+    var ctrl, $scope, $q, modalInstanceSpy, regionSpy, locationSpy, rightSpy;
+
+    beforeEach(module('access-rights.models', 'access-rights.controllers'));
+    beforeEach(function () {
+      modalInstanceSpy = jasmine.createSpyObj('modalInstanceSpy', ['dismiss']);
+      regionSpy = jasmine.createSpyObj('regionSpy', ['getAll']);
+      locationSpy = jasmine.createSpyObj('locationSpy', ['getAll']);
+      rightSpy = jasmine.createSpyObj('rightSpy', ['getRegions', 'getLocations',
+        'saveRegions', 'saveLocations', 'deleteByIds'
+      ]);
+    });
+    beforeEach(inject(function (_$controller_, _$rootScope_, _$q_) {
+      CRM = {
+        vars: {
+          summaryPage: {
+            summaryPage: '1'
+          }
+        }
+      };
+      $scope = _$rootScope_.$new();
+      $q = _$q_;
+
+      function getEntityMock(sequential, isRegion) {
```

Comment:
This `beforeEach` block is too long (and there is a function definition in the middle of the implementation that kinda disrupts the flow), i think you could refactor to isolate this spies initialization logic

```js
beforeEach(function () {
  $scope = _$rootScope_.$new();
  $q = _$q_;

  initSpies();

  ctrl = _$controller_('AccessRightsModalCtrl', {});
  $scope.$digest();
}));

/**
 *
 *
 */
function initSpies($q) {
  modalInstanceSpy = jasmine.createSpyObj('modalInstanceSpy', ['dismiss']);
  regionSpy = jasmine.createSpyObj('regionSpy', ['getAll']);
  locationSpy = jasmine.createSpyObj('locationSpy', ['getAll']);
  rightSpy = jasmine.createSpyObj('rightSpy', ['getRegions', 'getLocations',
    'saveRegions', 'saveLocations', 'deleteByIds'
  ]);

  [regionSpy.getAll, locationSpy.getAll].forEach(function (fn, idx) {});
  [rightSpy.getRegions, rightSpy.getLocations].forEach(function (fn, idx) {});
  [rightSpy.saveRegions, rightSpy.saveLocations].forEach(function (fn) {});
}

/**
 *
 *
 */
function getEntityMock(sequential, isRegion) {
  var difference = isRegion ? 0 : 10;
  return {
    id: sequential + difference,
    entity_id: sequential * 100 + difference
  };
}
```